Own root server with TCP only full recursion test

wtoorop opened the pull request.

wtoorop committed:
Own root server with TCP only full recursion test

saradickinson commented on src/test/check_getdns_context_set_dns_transport.h:
```diff
@@ -250,6 +250,8 @@
        struct getdns_context *context = NULL;
        struct getdns_dict *response = NULL;
        struct getdns_dict *extensions = getdns_dict_create();
+       struct getdns_list *root_servers = getdns_list_create();
```
This looks good - thanks. Is it worth just adding a comment here as to _why_ this is done so we don't forget in future?

wtoorop committed:
Explain own root server necessity in transport_7

Also query for the consistently large record and not the DNSKEY (which is sometimes large and sometimes not)
